ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -261,6 +274,8 @@ private boolean validateTxnList() throws CommandProcessorException {
 
             String userFromUGI = DriverUtils.getUserFromUGI(driverContext);
             driverContext.getTxnManager().openTxn(context, userFromUGI, driverContext.getTxnType());
+            // this is used by the test framework to more easily generate conflicting transactions
+            if (testConflictCB != null) testConflictCB.run();
```

Comment:
Strong NO for this approach. Use mocking techniques and refactor the code if needed. There shouldn't be any test hacks in the production code.

Reply:
Thanks for the feedback.
I re-wrote the test using mocking techniques, it turned out much better.
The approach I took was mostly because I didn't have strong enough experience with mocking style tests to understand I could do it this way.

Reply:
looks good!